### ISSUE TITLE
Query about the pre-training process

### ISSUE THREAD

**Issue by qwerhk839** [gpt4vision/OvSGTR#14]
Thank you for your outstanding work, but I still met many problems in the process of reproducing the pre-training results.

I use the following command to pre-train the groundingdino_swint:

bash scripts/DINO_train_dist.sh coco ./config/GroundingDINO_SwinT_OGC_pretrain.py .Datasets/ovsgtr_data  ./log/ovsgtr_vg_swint_pretrain ./GroundingDINO/weights/groundingdino_swint_ogc.pth

There may be some errors in your project:
1) Config file **GroundingDINO_SwinT_OGC_pretrain.py**: The **has_bbox_supervision** should be **False**. If it is True, it will cause an error in Line 116 in matcher.py, due to missing the "boxes" key. Besides, it will compute the **boxes loss** in Line 962 in **groundingdino.py**
2) Dataset file **coco.py**: Line 491 should be "while len(target['relations']) == 0:" and Line 494 should be removed, due to the missing key "boxes" in your provided **coco_train2017_triple.json**
3) Loss file **losses.py**:  Line 734 should be the following code, also due to the missing "boxes" key.
_if not self.rln_pretraining:
    num_boxes = sum(len(t["boxes"]) for t in targets)
else:
    num_boxes = sum(len(t["gt_names"]) for t in targets)_

**By fixing the mentioned errors, I can run the pretraining comment (I provided above) successfully. However, I cannot reproduce your results in Table 5.** **_without bounding box supervision_**.
May I ask about the implementation details of table 5 (pretraining command and the test command, as well as the required computing resources, e.g., how many 3090), and whether the code repair is correct?

Looking forward to your reply. THX :)

**Comment by JosephChenHub** [comment]
> Thank you for your outstanding work, but I still met many problems in the process of reproducing the pre-training results.
> 
> I use the following command to pre-train the groundingdino_swint:
> 
> bash scripts/DINO_train_dist.sh coco ./config/GroundingDINO_SwinT_OGC_pretrain.py .Datasets/ovsgtr_data ./log/ovsgtr_vg_swint_pretrain ./GroundingDINO/weights/groundingdino_swint_ogc.pth
> 
> There may be some errors in your project:
> 
> 1. Config file **GroundingDINO_SwinT_OGC_pretrain.py**: The **has_bbox_supervision** should be **False**. If it is True, it will cause an error in Line 116 in matcher.py, due to missing the "boxes" key. Besides, it will compute the **boxes loss** in Line 962 in **groundingdino.py**
> 2. Dataset file **coco.py**: Line 491 should be "while len(target['relations']) == 0:" and Line 494 should be removed, due to the missing key "boxes" in your provided **coco_train2017_triple.json**
> 3. Loss file **losses.py**:  Line 734 should be the following code, also due to the missing "boxes" key.
>    _if not self.rln_pretraining:
>    num_boxes = sum(len(t["boxes"]) for t in targets)
>    else:
>    num_boxes = sum(len(t["gt_names"]) for t in targets)_
> 
> **By fixing the mentioned errors, I can run the pretraining comment (I provided above) successfully. However, I cannot reproduce your results in Table 5.** **_without bounding box supervision_**. May I ask about the implementation details of table 5 (pretraining command and the test command, as well as the required computing resources, e.g., how many 3090), and whether the code repair is correct?
> 
> Looking forward to your reply. THX :)

The **has_bbox_supervision** should be **True** during pre-training , and the grounding process can be found at https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py

**Comment by myukzzz** [comment]
I have the same question: due to missing the "boxes" key in the provided json file(captions_train2017_triple. json and ,  captions_val2017_triple.json), has_bbox_supervision can only set to False.

On the other hand, in the https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py,  box is also predicted rather than included in the tag file, which is inconsistent with has_bbox_supervision=True

**Comment by JosephChenHub** [comment]
> I have the same question: due to missing the "boxes" key in the provided json file(captions_train2017_triple. json and , captions_val2017_triple.json), has_bbox_supervision can only set to False.
> 
> On the other hand, in the https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py, box is also predicted rather than included in the tag file, which is inconsistent with has_bbox_supervision=True

as mentioned above, has_bbox_supervision=True after grounding

**Comment by myukzzz** [comment]
In this case, the provided json file is not the result generated by grounding (without bounding box supervision), can you provide the correct json flie for pre-training?

**Comment by JosephChenHub** [comment]
> In this case, the provided json file is not the result generated by grounding (without bounding box supervision), can you provide the correct json flie for pre-training?

please use https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py to generate boxes.

**Comment by qwerhk839** [comment]
> > In this case, the provided json file is not the result generated by grounding (without bounding box supervision), can you provide the correct json flie for pre-training?
> 
> please use https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py to generate boxes.

Can you provide the complete run command, due to the "train_ds = ImgDataset(sys.argv[1], sys.argv[2], is_coco='coco' in sys.argv[2])", use python language_sgg_grounding.py directly will get an error.

**Comment by qwerhk839** [comment]
> > In this case, the provided json file is not the result generated by grounding (without bounding box supervision), can you provide the correct json flie for pre-training?
> 
> please use https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py to generate boxes.

Could you also provide the log file during pretraining? I want to ensure I have aligned the performance with your pre-trained ckpt. THX! :)

**Comment by ha1ha2hahaha** [comment]
> > > In this case, the provided json file is not the result generated by grounding (without bounding box supervision), can you provide the correct json flie for pre-training?
> > 
> > 
> > please use https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py to generate boxes.
> 
> Could you also provide the log file during pretraining? I want to ensure I have aligned the performance with your pre-trained ckpt. THX! :)

I think this py just need give the  coco path and  captions_train2017_triple.json path  in ImgDataset，But It aslo have error in the  captions_train2017_triple.json path，some annotations are not have relation turples，https://github.com/gpt4vision/OvSGTR/blob/master/tools/language_sgg_grounding.py#L158
